cmd/operator/main.go

```diff
@@ -99,6 +100,15 @@ func main() {
 		logger.Error(err, "loading kubeconfig failed")
 		os.Exit(1)
 	}
+	switch *mode {
```

Comment:
Only other thought I had about this, is that since it's being passed into the `operator.Options`, you could move this check to the [`defaultAndValidate`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/51e169c405cf3f88a7036bb778ebe718759166c2/pkg/operator/operator.go#L148) call.